crates/polars-plan/src/dsl/function_expr/array.rs

```diff
@@ -149,6 +153,10 @@ impl From<ArrayFunction> for SpecialEq<Arc<dyn ColumnsUdf>> {
     }
 }
 
+pub(super) fn length(s: &Column) -> PolarsResult<Column> {
+    Ok(s.array()?.array_lengths().into_column())
```

Comment:
Here we can:

```
let array = s.array()?;
let c = Column::new_scalar(array.name().clone(), array.width().into(), s.height());

if let Some(_) = s.validity() {
    // Handle validity
}
```

This way we don't need to allocate the actual memory for the lengths.